server/server.js

```diff
@@ -201,7 +201,7 @@ app.use(express.static(REACT_BUILD_DIR));
 
 
 
-
+//get all users to info to get emails? to get id? 
 app.get('/api/all/nutrition', cors(), async (req, res) => {
     try {
```

Comment:
So I would recommend making some changes in this code in the spots where you are actually looking for feedback, but based on what you said in the PR description, you can check if the user is logged in by checking if the user is logged in before doing anything else. So here you could do something like [the code you have above](https://github.com/KishaMims/Git-Healthy-App-/pull/3/files#diff-8cf1ffe3788af127768748703e2f15dcfb3a05ffd20b4c2375223637e3260938R161)

```
app.get('/api/all/nutrition', cors(), async (req, res) => {

  if (!req.oidc.isAuthenticated()) { // user is not logged in so return an error
    res.status(401).json({ error: 'user not logged in '})
  }

  const current_user = await db.query('SELECT * FROM users WHERE email = $1', req.oidc.user.email);
  const user_meals = await db.query('SELECT * FROM meal where user_id = $1', current_user.id);

   return res.json(user_meals);
}
```

Anyways this is just a rough idea but hopefully this gives you something to go off of. 

A couple key points:

1) you can ensure the user is logged in by calling `req.oidc.isAuthenticated()` first
2) you should use the email from `req.oidc.user.email` for security reasons (I can explain in 1:1)
3) once you have the user email, you need to look up the user object in your db so you can get the user id which you will need for future queries, like I do in my code snippet.

Another question: do you have your schema checked in? I was looking at `db.sql` [here](https://github.com/KishaMims/Git-Healthy-App-/blob/main/server/db.sql) and I just see the old schema stuff. Nothing for defining your users/meals/nutrition tables so it was a little hard for me to write queries which I am sure will work because I'm not sure exactly what your schema is.

Hopefully this gives you something to go off of/ask more questions in tech-help, etc tomorrow!